dds/DCPS/transport/rtps_udp/RtpsUdpDataLink.cpp

```diff
@@ -3808,6 +3860,8 @@ RtpsUdpDataLink::RtpsWriter::gather_heartbeats(OPENDDS_VECTOR(TransportQueueElem
     }
     readers_expecting_heartbeat_.clear();
   }
+
+  return !(additional_guids.empty() && preassociation_readers_.empty() && lagging_readers_.empty() && readers_expecting_heartbeat_.empty());
```

Comment:
Not sure it would matter, but this might be better as `!a.empty() || !b.empty()...` for short-circuiting